### PR TITLE
Rename `OSM.intersection` and `OSM.road`

### LINKED ISSUE #593: Rename `OSM.intersection` and `OSM.road`
I realised the names are misnomers. They no longer return intersections/roads. What about `OSM.nearest_node` and `OSM.nearest_edge`?

### PR DESCRIPTION
**Pull request by AayushSabharwal**
Closes #593

### CODE REVIEW

File: docs/src/api.md [hunk 1]
```diff
@@ -15,6 +16,7 @@ OSMAgent
 ```
 
 ## Agent/model retrieval and access
+
```
Comment (Datseris) [inline docs/src/api.md:19]: Why have you added all these empty lines between headers and docs? they are not necessary, right? everything compiles without them.

**Review by Datseris — CHANGES_REQUESTED** [review]
Please add these breaking changes in the CHANGELOG.md. Probably also in deprecations.jl.

File: docs/src/api.md (hunk 1 above)
Comment (AayushSabharwal) [reply docs/src/api.md:19]: I forgot about that. I installed a markdown autoformatter recently, and I forgot that it was enabled and formats documents on save. I'll remove these lines

**Comment by AayushSabharwal** [comment]
I've fixed the line spacing, but am still unable to figure out `@deprecate`. I'll add to the changelog in #591 since adding it here would possibly cause a merge conflict.

**Comment by AayushSabharwal** [comment]
There's something wrong with docs...

**Comment by Datseris** [comment]
> There's something wrong with docs... Both here and https://github.com/JuliaDynamics/Agents.jl/pull/591

Don't worry about docs for now, I still need to merge OSM plotting in InteractiveDynamics.jl before re-doing the docs anyways. Just be sure we have the deprecations for the changed names in place and all is good!

**Pull request merged**